src/main/java/com/github/princesslana/eriscasper/util/Shard.java

```diff
@@ -0,0 +1,49 @@
+package com.github.princesslana.eriscasper.util;
+
+import com.github.princesslana.eriscasper.ErisCasperFatalException;
+import com.github.princesslana.eriscasper.data.immutable.Wrapper;
+import com.ufoscout.properlty.Properlty;
+import java.util.Optional;
+
+public class Shard implements Wrapper<Integer[]> {
```

Comment:
I don't think implementing Wrapper here is semantically correct. I think just having a `toArray` method is cleaner.

Reply:
Have you considered using immutables and making this a `@Tuple`? You could still do the checks on the values by creating our own `Shard.of` factory method.

Reply:
Implementing Wrapper allows it to be used with jackson properly. It also is correct in the sense that it's wrapping an array of integers. If I don't use Wrapper Jackson may not like it so much. If you know how to get it to where Jackson treats it like an Integer[] like it does with Wrapper I can replace it.

Reply:
Except we don't wrap an array of integers, we have two integer fields.

The Jackson magic you are looking for is `@JsonValue`. If you annotate the `toArray` method with it, then Jackson will use that to serialize.

Reply:
Alright, I'll use that then.